wallet/client.go

```diff
@@ -39,16 +43,30 @@ func (c *grpcClient) SetServerAddrs(servers []string) {
 	c.servers = servers
 }
 
+func (c *grpcClient) SetTimeoutConnection(timeout time.Duration) {
+	if timeout > 0 {
+		c.timeout = timeout
+	}
+}
+
 func (c *grpcClient) connect() error {
 	if c.conn != nil {
 		return nil
 	}
 
 	for _, server := range c.servers {
-		conn, err := grpc.NewClient(server,
-			grpc.WithTransportCredentials(insecure.NewCredentials()))
+		opts := make([]grpc.DialOption, 0)
+		opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
+
+		if len(c.servers) > 1 {
+			opts = append(opts, grpc.WithContextDialer(func(_ context.Context, s string) (net.Conn, error) {
+				return net.DialTimeout("tcp", s, c.timeout)
+			}))
+		}
+
+		conn, err := grpc.NewClient(server, opts...)
 		if err != nil {
-			continue
+			return err
```

Comment:
Here we must continue, we have a list of servers...
We can't stop the whole process if one of the servers is offline.